pyingest/parsers/aasnova.py

```diff
@@ -0,0 +1,136 @@
+## Convert month name to number
+mnth2num = {month.lower(): index for index, month in enumerate(calendar.month_abbr) if month}
+## Exceptions
+class NoPublicationDateFound(Exception):
+    pass
+
+class AASNovaParser(BaseRSSFeedParser):
+
+    def extract_data(self, entry):
+        links = {}
+        # Get the categories
+        try:
+            categories = [c.text for c in entry.find_all('category')]
+        except:
+            categories = []
+        check = [c for c in categories if re_cats_allowed.search(c)] 
+        # Skip if the entry is not the right category
+        if len(check) == 0:
+            return {}
+        keywords = []
+        for c in categories:
+            kmat = re_CDATA.search(c)
+            if kmat:
+                keywords.append(kmat.group('tag'))
+        # Remove AAS Nova tags
+        keywords = [k for k in keywords if k not in AASNova_tags]
+        # Start gathering the necessary fields
+        title = entry.find('title').text
+        # Get author data
+        au = entry.find('dc:creator').text
+        amat = re_CDATA.search(au)
+        if amat:
+            au = amat.group('tag')
+        if au.lower() != 'astrobites':
+            audata = au.split()
+            fname = audata[0]
+            lname = " ".join(audata[1:])
+            author = "%s, %s" % (lname, fname)
+        else:
+            author = "Astrobites"
+            lname = "Astrobites"
+        # Nova record number
+        nova_entry = entry.find('guid').text
+        links['ELECTR'] = nova_entry
+        # The record number follows from the URL:
+        # e.g. https://aasnova.org/?p=10858
+        # Of course, we can parse the URL using the urllib.parse library
+        # but that seems overkill here
+        recno = nova_entry.split('p=')[-1]
+        # Attempt to find the citation in the record (for the paper
+        # discussed in this AAS Nova entry)
+        try:
+            citation = entry.find('citation').text
+        except:
+            citation = None
+        # If we found a citation, see if we can capture a DOI
+        doi = None
+        if citation:
+            dmat = re_doi.search(citation)
+            if dmat:
+                doi = dmat.group('doi').strip()
+        # Get the publication date
+        try:
+            pdate = entry.find('pubdate').text.split()
+            pubdate = " ".join(pdate[1:4])
+            mnth = mnth2num.get(pdate[2].lower(), '00')
+            year = pdate[3]
+        except:
+            raise NoPublicationDateFound("AAS Nova entry does not contain publication date")
+        # Get the journal entry
+        journal = "AAS Nova Highlight %s, id. %s" % (pubdate, recno)
+        # Get the abstract data
+        try:
+            abstract = entry.find('description').text.replace('<![CDATA[','').replace(']]>','').strip()
+            abstract = re.sub('The post.*?appeared first on AAS Nova\.','', abstract).strip()
```

Comment:
This line is throwing a DeprecatedWarning in pytest because of the "...AAS Nova\."  The slash preceding the period is not required.  Related possibly, should the "re.sub('The post.*?..." be "re.sub(r'The post.*?" with an r preceding the single quote to indicate its supposed to use regex interpreter?

Reply:
I tested my hypothesis -- there's no warning if you have the slash but precede the regex with r.  You have all your other compile regexes with the r'', so I would do the same with the re.sub.